### ISSUE TITLE
Change fonts of widgets, including text widget

### ISSUE THREAD

**Issue by georgtree** [aplsimple/pave#7]
Hello!
I know it is a dumb question, but I can't change the global font of widgets, I tried to use this:
```tcl
  pave basicDefFont "-family {Fira Code} -size 15 -weight normal -slant roman -underline 0 -overstrike 0"
  pave basicTextFont "-family Fira Code} -size 15 -weight normal -slant roman -underline 0 -overstrike 0"
```

and this:
```tcl
  pave basicDefFont {Fira Code}
  pave basicTextFont {Fira Code}
```
But font doesn't change...
Thank you in advance!

**Comment by aplsimple** [comment]
I can only share with my font use in alited.

Mostly it's about alited/src/ini.tcl, so you can extract all "font" instances from it to see how it works.

If I were you, I would mostly use ini::InitFonts procedure, namely its lines:

```
foreach font {TkDefaultFont TkMenuFont TkHeadingFont TkCaptionFont} {
  font configure $font {*}$al(FONT)
}
foreach font {TkSmallCaptionFont TkIconFont TkTooltipFont} {
  font configure $font {*}$smallfont
}
```
where $al(FONT) means "default font of app", $smallfont means "font for small sized messages like status bar, tooltip etc."

Of course, alited provides its own settings for fonts, with separate font sizes for different uses, plus the text font. Still it's not too complex to dig it out.

**Comment by georgtree** [comment]
Hello!
Still struggling with fonts settings.

The listbox doesn't response to font settings, I supply parameter:
```tcl
-font {-family "Fira Code" -size 10 -weight normal -slant roman -underline 0 -overstrike 0}
```

**Comment by georgtree** [comment]
i also tried to redefine default font and try `-font TkDefaultFont`, and still no effect, I see that in alited code you set:
```tcl
{.fraBot.fra.LbxInfo - - - - {pack -side left -fill both -expand 1}
      {-h 1 -w 40 -lvar ::alited::info::list -font $al(FONT,defsmall)
      -highlightthickness 0 -onevent {
      <<ListboxSelect>> "alited::info::ListboxSelect %w"
      <ButtonPress-3> "alited::info::PopupMenu %X %Y"}}}
```
and it works, don't understand why my code fails...

**Comment by georgtree** [comment]
At the same time, labels changes font, tablelist works for body elements

**Comment by georgtree** [comment]
Also I added this:
```tcl
foreach font {TkDefaultFont TkMenuFont TkHeadingFont TkCaptionFont TkSmallCaptionFont TkIconFont TkTooltipFont} {
        font configure $font {*}$globVars(customFont)
    }
```
Looks like I missed some important point there...

**Comment by georgtree** [comment]
text widget also doesn't respond, only work if font is defined as an option to hl_tcl package

**Comment by georgtree** [comment]
Entry and ttk::combobox widgets also doesn't change font

**Comment by aplsimple** [comment]
Try untouchWidgets method of the apave object, to disable theming separate widgets, (just after creating the object) e.g.

```
set myobj ::myns::pavedObj
apave::APave create $myobj .mywin
$myobj untouchWidgets *.lbx1 *.tex2
```

So that the widgets with names matching the glob patterns `*.lbx1 *.tex2` will not be themed by apave. At that, you should handle theming them yourself.

By default, the non-ttk widgets are themed, to be in accordance with the current theme.

Another way is using `after idle` commands to theme those widgets, after a dialog is built and before going to tkwait cycle.

As for combobox, it's the most hard widget of all, as for customisation. It needs an individual solution to get the success. Listbox is better, up to customizing rows -fg and -bg.

**Comment by aplsimple** [comment]
In the layout, -afteridle option could help too. For example:
```
{.lbx + T 1 1 {-st nswe} {-lvar ::t::lvar -h 4 -w 30 -lbxsel dark -ALL yes -afteridle {%w config -font Times}}}
```
Here %w stands for the listbox (lbx) path.

**Comment by georgtree** [comment]
Hello!
This untouchWidgets worked magically, now all picks my custom font, thank you very much!

**Comment by georgtree** [comment]
But what if I want to change only font in the theme that alited provided, and save the rest of customization?

**Comment by georgtree** [comment]
The only thing I can't change font with `-afteridle {%w config -font TkDefaultFont}` is the header of tablelist. The only solution is to drop the apave theming, but it is difficult because they are my base widgets.
Is there any other solution?

**Comment by aplsimple** [comment]
It's about *-labelfont* option of tablelist widget, seemingly. So that you might try
```
-afteridle {tablelist::tablelist %w -labelfont Times; other-config-commands-for-%w}
```

See tablelist description:

https://www.nemethi.de/tablelist/tablelistWidget.html#labelfont

**Comment by georgtree** [comment]
It works, thank you!
Tablelist is really complex  thing, I lost a little bit in it :(

**Comment by georgtree** [comment]
Last question - do you know if it is possible to change the font of top menu in Windows?

![Image](https://github.com/user-attachments/assets/ba4b5a8c-8611-4773-a237-d8cefd6803f7)
As far as I understand, we can't do much with that menu, but maybe there is a way to change font? I was able to change the font for
drop-down menu's points. Thank you.

**Comment by aplsimple** [comment]
No, Windows' menu bars are themed on the OS level only. If Tcl/Tk code contains some customisation of menu bar, in Windows it is just ignored, depart from Linux.

The menu items (incl. popups ) are quite customizable.

**Comment by georgtree** [comment]
Got it, thank you.
Interesting, is there a way to emulate this widget with another widget that allows customization...

**Comment by aplsimple** [comment]
You can do hamburger button and show under it a popup menu. It's a fashion of these days.
